### PR TITLE
add styleChecker Action

### PR DESCRIPTION
**Pull request by Murmele**
do not sort includes, because sometimes the order is important (Scintilla)

### CODE REVIEW

**Comment by Murmele** [comment]
#!/bin/bash

# Variable that will hold the name of the clang-format command
FMT=""

# Some distros just call it clang-format. Others (e.g. Ubuntu) are insistent
# that the version number be part of the command. We prefer clang-format if
# that's present, otherwise we work backwards from highest version to lowest
# version.
for clangfmt in clang-format{,-{4,3}.{9,8,7,6,5,4,3,2,1,0}}; do
    if which "$clangfmt" &>/dev/null; then
        FMT="$clangfmt"
        break
    fi
done

# Check if we found a working clang-format
if [ -z "$FMT" ]; then
    echo "failed to find clang-format"
    exit 1
fi

function format() {
    for f in $(find $@ -name '*.h' -or -name '*.m' -or -name '*.mm' -or -name '*.c' -or -name '*.cpp'); do 
        echo "format ${f}";
        ${FMT} -i ${f};
    done

    echo "~~~ $@ Done ~~~";
}

# Check all of the arguments first to make sure they're all directories
for dir in "$@"; do
    if [ ! -d "${dir}" ]; then
        echo "${dir} is not a directory";
    else
        format ${dir};
    fi
done

**Comment by Murmele** [comment]
@stefanknotzer what do you think about?

**Comment by stefanknotzer** [comment]
Great job! :+1:
I like that the source files are getting shorter (less lines).

**Comment by Murmele** [comment]
@stefanknotzer can you approve and then mergin?

**Comment by stefanknotzer** [comment]
I suggest merging and closing as many pull requests pending as possible before formatting the code.
I have troubles approving `.clang-format` with clang 11 - which clang version are you using?

**Comment by Murmele** [comment]
In the ci it is 13. Yes I agree with you

**Review by stefanknotzer — APPROVED** [review]
Approved with clang 13 on Windows 10

**Comment by stefanknotzer** [comment]
I just stumbled over a switch case statement with braces in the case (TreeModel.cpp).
Setting `IndentCaseLabels: true` would be nice.

IndentCaseLabels

File: .clang-format [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+    SortPriority:    0
+    CaseSensitive:   false
+  - Regex:           '.*'
+    Priority:        1
+    SortPriority:    0
+    CaseSensitive:   false
+IncludeIsMainRegex: '(Test)?$'
+IncludeIsMainSourceRegex: ''
+IndentAccessModifiers: false
+IndentCaseLabels: false
```
Comment (Murmele) [inline .clang-format:98]:
```suggestion
IndentCaseLabels: true
```

**Comment by exactly-one-kas** [comment]
What do you think about also formatting CMake files?

```
name: Check CMake code style
run: |
  pip install cmake-format==0.6.13 && \
  find \( -type d -path './dep/*/*' -prune \) -o \( -name CMakeLists.txt -exec cmake-format --check {} + \)
```

**Comment by Murmele** [comment]
> What do you think about also formatting CMake files?
> 
> ```
> name: Check CMake code style
> run: |
>   pip install cmake-format==0.6.13 && \
>   find \( -type d -path './dep/*/*' -prune \) -o \( -name CMakeLists.txt -exec cmake-format --check {} + \)
> ```

Good idea, just add it to this branch. It should be possible or?

**Comment by Murmele** [comment]
@exactly-one-kas @stefanknotzer  I would like to get the following pr into then we can apply once the format and merging this one or would you like to add another merge request first?
- #122 
- #120 
- #63

**Comment by exactly-one-kas** [comment]
> would you like to add another merge request first?

No, that list is fine by me

**Comment by Murmele** [comment]
@exactly-one-kas I added the cmake formatting also to the .cl-fmt file.

**Review by exactly-one-kas — APPROVED** [review]

**Pull request merged**